### PR TITLE
Add post buff hook

### PR DESCRIPTION
**Pull request by erickgalinkin**
Add untransform to buffs.base (placeholder) and low_resource_languages.py. Add post_buff_hook state tracking in Probe base class to flag the need to call _postprocess_buff, which calls buff.untransform on the attempt.

### CODE REVIEW

File: garak/buffs/base.py [hunk 1]
```diff
@@ -72,6 +73,12 @@ def transform(
         """attempt copying is handled elsewhere. isn't that nice"""
         yield self._derive_new_attempt(attempt)
 
+    def untransform(
+        self, attempt: garak.attempt.Attempt
+    ) -> garak.attempt.Attempt:
+        """Only necessary if buff requires a post buff hook"""
+        pass
```
Comment (leondz) [inline garak/buffs/base.py:80]: seeing as post_buff_hook gates calling untransform, how do you feel about raising a `NotImplementedError` here, or returning  `attempt`?

File: garak/probes/base.py [hunk 2]
```diff
@@ -70,12 +71,21 @@ def _buff_hook(
             return attempts
         buffed_attempts = []
         for buff in _config.transient.buff_instances:
+            if buff.post_buff_hook:
+                self.post_buff_hook = True
```
Comment (leondz) [inline garak/probes/base.py:75]: is the behavior to call untransform for all buffs, as soon as one with post_buff_hook asserted is encountered?

Comment (erickgalinkin) [reply garak/probes/base.py:75]: There's probably a more efficient way to do this, but I don't know what it is.

**Review by leondz — APPROVED** [review]
lgtm

**Pull request merged**